File: twosum.java
import java.util.Arrays;
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int temp = target - nums[i];
            if (map.containsKey(temp)) {
                int[] result = {map.get(temp), i};
                return result;
            }
            map.put(nums[i], i);
        }
        return new int[0];
    }

    public static void main(String[] args) {
        int[] nums = {1,2,3,4,5,6};
        Solution solution = new Solution();
        int[] result = solution.twoSum(nums, 5);
        System.out.println(Arrays.toString(result));
    }
}

File: twosum.py
from pip import List

class Solution:        
    # constructor
    __slots__ = ["__nums", "__target"]

    def __init__ (self, nums, target):
        # state        
        self.__nums = nums
        self.__target = target
        
    # method
    def two_sum (self):
        map = {}
        for i in range(0, len(self.__nums)):
            if self.__target - self.__nums[i] in map:
                return [i, map[self.__target-self.__nums[i]]]
            map[nums[i]] = i 
            
        return [] # not found!
    
if __name__ == "__main__":
    nums = [1,2,3,4,5,6]
    solution = Solution(nums, 5)
    x = solution.two_sum()
    print(x)